lazyflow/utility/reorderAxesDecorator.py

```diff
@@ -0,0 +1,263 @@
+import functools
+from lazyflow.operators.opReorderAxes import OpReorderAxes
+
+
+def reorder_options(internal_axes_order, exceptions=[],
+                    output_axes_order='tczyx'):
+    """
+        Adds reorder options to the specific operator class that the decorator
+        reorder accesses. This is not done by handing arguments to the reorder
+        decorator directly, because then the reorder decorator would need to be
+        wrapped, which would lead return a function (wrapper) as the operator
+        class, which conflicts with inheritance of the specific operator (as
+        the inheriting operator class appears to be a function). Therefore the
+        arguments are outsourced in this decorator.
+
+        note: Use this function decorator only in combination with the reorder
+              decorator and as second (inner) decorator.
+              (see example usage below)
+
+        @param internal_axes_order: str Axes order the decorated operator
+                                        assumes internally.ResizedShape
+        @param exceptions: list(str) List of slot names that do not need to be
+                                     reordered.
+        @param output_axes_order: str Output Axes order for all reordered
+                                      output slots.
+
+        Example usage:
+
+            @reorder
+            @reorder_options('tzyxc', ['ResizedShape'])
+            class OpResize5D(Operator):
+                ...
+
+    """
+    assert isinstance(internal_axes_order, str)
+    assert isinstance(exceptions, list)
+    assert all([isinstance(ex, str) for ex in exceptions])
+    assert isinstance(output_axes_order, str)
+
+    def add_exceptions(cls):
+        assert all([ex in [s.name for s in cls.inputSlots + cls.outputSlots]
+                    for ex in exceptions])
+        cls._internal_axes_order = internal_axes_order
+        cls._reorder_exceptions = exceptions
+        cls._output_axes_order = output_axes_order
+        return cls
+
+    return add_exceptions
+
+
+def guard_methods(cls):
+    """
+        helper function for reorder
+    """
+    methods = [member for member, member_type in cls.__dict__.items()
+               if isinstance(member_type, type(lambda:0)) and
+               member != '__getattribute__' and  # will be adapted individually
+               member != '__init__'  # will be adapted individually
+               ]
+
+    class_methods = [member for member, member_type in cls.__dict__.items()
+                     if isinstance(member_type, classmethod)]
+
+    property_methods = [member for member, member_type in cls.__dict__.items()
+                        if isinstance(member_type, property)]
+
+    # flag when methods are called from within the class
+    cls._inner_call = 0
```

Comment:
is `cls._inner_call` expected to take numerical values? What is this indicating exactly? As far as I can see it's only ever set to `True` in the wrapped `__init__` in order to access the original slots... right?